SoftLayer/managers/ordering.py

```diff
@@ -471,3 +473,22 @@ def package_locations(self, package_keyname):
 
         regions = self.package_svc.getRegions(id=package['id'], mask=mask)
         return regions
+
+    def get_location_id(self, location):
+        """Finds the location ID of a given datacenter
+
+        This is mostly used so either a dc name, or regions keyname can be used when ordering
+        :param str location: Region Keyname (DALLAS13) or datacenter name (dal13)
+        :returns: integer id of the datacenter
+        """
+
+        mask = "mask[id,name,regions[keyname]]"
+        if match(r'[a-zA-Z]{3}[0-9]{2}', location) is not None:
```

Comment:
I might be wrong, but there could still be a case where a user is passing in an id as the location, and previously this was just passed on to the order.

For backwards compatibility, I think this method may want to check for a digit-only string (`isdigit`) and short circuit the lookup if that is the case.